app/src/main/java/nu/yona/app/utils/AppUtils.java

```diff
@@ -288,25 +289,37 @@ public static String getHourDigit(String digit)
 	 */
 	public static void reportException(String className, Exception e, Thread t, DataLoadListener listener)
 	{
-		ErrorMessage errorMessage = new ErrorMessage();
-		if (e != null && e.getMessage() != null)
+		ErrorMessage errorMessage = getErrorMessage(e);
+		if (listener != null)
 		{
-			errorMessage.setMessage(e.getMessage());
+			listener.onError(errorMessage);
 		}
 		else
 		{
-			errorMessage.setMessage(YonaApplication.getAppContext().getString(R.string.error_message));
+			showErrorToast(errorMessage);
+			logExceptionToCrashlytics(e);
 		}
-		if (listener != null)
+	}
+
+	/**
+	 * getErrorMessage from give Exception or return a generic error message
+	 */
+
+	public static ErrorMessage getErrorMessage(Exception e)
+	{
+		ErrorMessage errorMessage = new ErrorMessage();
+		if (e != null && e.getMessage() != null)
 		{
-			listener.onError(errorMessage);
+			errorMessage.setMessage(e.getMessage());
+			Logger.loge(e.getClass().getSimpleName(), e.getMessage());
 		}
 		else
 		{
-			showErrorToast(errorMessage);
-			logExceptionToCrashlytics(e);
+			String message = YonaApplication.getAppContext().getString(R.string.generic_error_message);
+			errorMessage.setMessage(message);
+			Logger.loge(e.getClass().getSimpleName(), message);
```

Comment:
This line is nearly identical to the one I just commented on. It could easily be put below the  if/else and take the message text from the just constructed error message. Now it should be removed, as said in the previous comment.